### ISSUE TITLE
ref's repr is misleading

### ISSUE THREAD

**Issue by sorawee** [brownplt/pyret-lang#645]
In horizon:

```
> data Box: box(ref v) end
> ref-set(a.v, a)
> a.v
box(<cyclic-ref-1>)
> a!v
box(<cyclic-object-1>)
```

The difference between the repr of `a.v` and `a!v` is not clear. I think that it would be more useful to show
`ref box(<cyclic-ref-1>)` for `a.v`.

Also, what is the warning icon here?

<img width="167" alt="screenshot 2016-02-13 18 37 18" src="https://cloud.githubusercontent.com/assets/9099577/13030693/63927522-d281-11e5-886e-571496c8f7d0.png">

Lastly, CPO couldn't render cyclic structure correctly. I presume that this is fixed in horizon (because it has no problem).

**Comment by blerner** [comment]
The warning icon, if you mouse over it, should give you alt text that says "This value may be stale; click to refresh" -- because it is a preview of mutable data.

The output you see above actually makes perfect sense...if you already know what the algorithm is doing.  `a.v` is a box, whose contents are a ref that points cyclically back to the original object being printed.  On the other hand `a!v` dereferences that box, so it contains an object that has already been printed.

**Comment by sorawee** [comment]
Yes, I don't what it does. The problem is, it would be nice to see the distinction between values and the box with values. Right now they are printed in a similar manner.

**Comment by sorawee** [comment]
Also, is there a better way than showing the icon? Perhaps blinking? I don't like icon because it's too big and it's not transparent, preventing users to see text behind the icon.

**Comment by blerner** [comment]
The design constraint here is to not obscure anything by default, and to not be annoying.  Blinking, flashing,etc, being visible by default break that.  Being large helps make it a clickable target, though.  I'm certainly open to a good UI designer helping us out here, since the design I put together for this (and for expandable images, with their magnifying glasses) was a first- or second-draft UI...
